tests/backends/test_dbscript.py

```diff
@@ -47,321 +50,523 @@ def timestwo(x):
 
-    ensemble.add_function("test_func", function=timestwo, device="CPU")
+    # Add script via function
+    ensemble.add_function(
+        "test_func",
+        function=timestwo,
+        device=test_device,
+        devices_per_node=test_num_gpus
+    )
 
-    # Ensemble must add all available DBScripts to new entity
+    # Add an additional ensemble member and attach a script to the new member
     ensemble.add_model(smartsim_model)
-    smartsim_model.add_script("test_script2", script=torch_script_str, device="CPU")
+    smartsim_model.add_script(
+        "test_script2",
+        script=torch_script_str,
+        device=test_device,
+        devices_per_node=test_num_gpus
+    )
 
     # Assert we have added both models to the ensemble
     assert len(ensemble._db_scripts) == 2
+
     # Assert we have added all three models to entities in ensemble
     assert all([len(entity._db_scripts) == 3 for entity in ensemble])
 
-    exp.start(db, ensemble, block=True)
-    statuses = exp.get_status(ensemble)
-    exp.stop(db)
-    assert all([stat == status.STATUS_COMPLETED for stat in statuses])
+    try:
+        exp.start(db, ensemble, block=True)
+        statuses = exp.get_status(ensemble)
+        assert all([stat == status.STATUS_COMPLETED for stat in statuses])
+    finally:
+        exp.stop(db)
 
 
 @pytest.mark.skipif(not should_run, reason="Test needs Torch to run")
-def test_colocated_db_script(fileutils, wlmutils):
+def test_colocated_db_script(fileutils, wlmutils, mlutils):
     """Test DB Scripts on colocated DB"""
 
+    # Set the experiment name
     exp_name = "test-colocated-db-script"
-    exp = Experiment(exp_name, launcher="local")
 
-    # get test setup
+    # Retrieve parameters from testing environment
+    test_launcher = wlmutils.get_test_launcher()
+    test_interface = wlmutils.get_test_interface()
+    test_port = wlmutils.get_test_port()
+    test_device = mlutils.get_test_device()
+    test_num_gpus = mlutils.get_test_num_gpus()
     test_dir = fileutils.make_test_dir()
-    sr_test_script = fileutils.get_test_conf_path("run_dbscript_smartredis.py")
+    test_script = fileutils.get_test_conf_path("run_dbscript_smartredis.py")
     torch_script = fileutils.get_test_conf_path("torchscript.py")
 
-    # create colocated model
-    colo_settings = exp.create_run_settings(exe=sys.executable, exe_args=sr_test_script)
+    # Create the SmartSim Experiment
+    exp = Experiment(exp_name, launcher=test_launcher)
 
+    # Create RunSettings
+    colo_settings = exp.create_run_settings(exe=sys.executable, exe_args=test_script)
+
+    # Create model with colocated database
     colo_model = exp.create_model("colocated_model", colo_settings)
     colo_model.set_path(test_dir)
-    colo_model.colocate_db(
-        port=wlmutils.get_test_port(),
+    colo_model.colocate_db_tcp(
+        port=test_port,
         db_cpus=1,
         limit_app_cpus=False,
         debug=True,
-        ifname="lo",
+        ifname=test_interface,
```

Comment:
Same "I imagine `ifname="lo"` to be FAR more popular than `ifname=["ipogif0"]` for colocated dbs. Should we consider parameteriziing this test to continue to explicitly test against `"lo"`?" Thought here and for remaining tests in this file